### DISCUSSION TITLE
How to use full cpu cores when screen is off?

### DISCUSSION THREAD

**Discussion by hengwu0** [termux/termux-app#3019] [Q&A] [answered]
I have two phones, they are both android 6, but different manufacturers. when the screen is off, I found that only half cores of the cpu is used. each core is at 100% and the others not onlined. so it don't uses all cores. if i start more programs, it will become slower but still 50% idle cores. what can i do to use full cpu cores?
I already rooted the device, and there already had wakelock in `/sys/power/wake_lock` file. Here is a phone's detail:
```sh
Linux localhost 3.18.24-perf-g80fce17 #1 SMP PREEMPT Fri Dec 2 19:28:47 CST 2016 aarch64 aarch64 aarch64 GNU/Linux
[ro.build.soft.version]: [A.06]
[ro.build.version.all_codenames]: [REL]
[ro.build.version.base_os]: [OPPO/R9s/R9s:6.0.1/MMB29M/1390465867:user/release-keys]
[ro.build.version.codename]: [REL]
[ro.build.version.incremental]: [eng.root.20161202.191841]
[ro.build.version.opporom]: [V3.0]
[ro.build.version.ota]: [R9s_11.A.06_006_201612021852]
[ro.build.version.preview_sdk]: [0]
[ro.build.version.release]: [6.0.1]
[ro.build.version.sdk]: [23]
[ro.build.version.security_patch]: [2016-09-01]

```

**Comment by agnostic-apollo** [comment]
cpu shut off would be controlled by cpu hotplug like `mpdecision` and cpu governors in the kernel and would require tuning those to your desired settings. You can use a kernel tuner app like Kernel Auditor and possibly a custom kernel if your stock kernel does not allow modification or changing for cpu hotplugs to different driver. Would require research to understand and do this stuff. Some guides are available at xda and stackoverflow, etc.

https://forum.xda-developers.com/t/ref-guide-most-up-to-date-guide-on-cpu-governors-i-o-schedulers-and-more.3048957/

https://forum.xda-developers.com/t/ref-guide-most-up-to-date-guide-on-cpu-governors-i-o-schedulers-and-more.3048957/#post-67085267

https://forum.xda-developers.com/t/kernels-guide-kernels-explained-governors-hotplug-frequency-cpu-gpu-i-o-etc.3032171/

**Reply by hengwu0** [reply]
I executed
```sh
stop thermald  
stop mpdecision  
stop thermal-engine  
```
and seems not work...

**Comment by hengwu0** [comment, marked as answer]
OPPO use `persist.sys.enable.hypnus` to control device‘s sleep. It should use **this command** to disable oppo's hypnus.ko and to use android's default cpu core policy。
```sh
setprop persist.sys.enable.hypnus 0
```
